packages/hoppscotch-app/helpers/realtime/WSSessionAdapter.ts

```diff
@@ -0,0 +1,196 @@
+import { Ref } from "@nuxtjs/composition-api"
+import { NuxtApp } from "@nuxt/types/app"
+import { debounce } from "lodash"
+import { BehaviorSubject } from "rxjs"
+import { logHoppRequestRunToAnalytics } from "../fb/analytics"
+import { useStream } from "../utils/composables"
+import {
+  addWSLogLine,
+  addWSProtocol,
+  deleteWSProtocol,
+  setWSLog,
+  setWSProtocols,
+  setWSSocket,
+  updateWSProtocol,
+  WSLog$,
+  WSProtocols$,
+  WSSocket$,
+} from "~/newstore/WebSocketSession"
+
+export default class WSSessionAdapter {
+  error$: BehaviorSubject<string | null>
+  connecting$: BehaviorSubject<boolean>
+  connection$: BehaviorSubject<boolean>
+  urlValid$: BehaviorSubject<boolean>
+
+  socket: Ref<WebSocket>
+  worker!: Worker
+
+  log = useStream(WSLog$, [], setWSLog)
+  protocols = useStream(WSProtocols$, [], setWSProtocols)
+
+  toast: any
+  t: any
+
+  constructor(nuxt: NuxtApp, toast: any, t: any) {
```

Comment:
These things should not be here.

Our goal with the Adapter is to separate rendering logic from the connection logic.

There should not be any UI code (toasts, translations, nuxt usage) here.